### PR TITLE
expression: implement vectorized evaluation for builtinFindInSetSig

### PR DESCRIPTION
**Pull request by icditwang**
PCP #12103
### What problem does this PR solve? 
implement vectorized evaluation for builtinFindInSetSig

### What is changed and how it works?

```
BenchmarkVectorizedBuiltinStringFunc/builtinFindInSetSig-VecBuiltinFunc-8         	   21078	     56566 ns/op	   10640 B/op	     665 allocs/op

BenchmarkVectorizedBuiltinStringFunc/builtinFindInSetSig-NonVecBuiltinFunc-8      	   10000	    101829 ns/op	   10640 B/op	     665 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -778,11 +778,45 @@ func (b *builtinLpadSig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
...
+	}
+	strlist, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(strlist)
+	if err := b.args[1].VecEvalString(b.ctx, input, strlist); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
```
Comment (b41sh) [inline expression/builtin_string_vec.go]: we can use `result.MergeNulls` here

File: expression/builtin_string_vec.go [hunk 2]
```diff
@@ -778,11 +778,45 @@ func (b *builtinLpadSig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
...
+	res := result.Int64s()
+	for i := 0; i < n; i++ {
+		if str.IsNull(i) || strlist.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
+		strlistI := strlist.GetString(i)
+		if len(strlistI) == 0 {
+			res[i] = 0
```
Comment (b41sh) [inline expression/builtin_string_vec.go]: please add `continue`

File: expression/builtin_string_vec.go (hunk 1 above)
Comment (icditwang) [reply expression/builtin_string_vec.go]: done!

File: expression/builtin_string_vec.go [hunk 3]
```diff
@@ -778,11 +778,43 @@ func (b *builtinLpadSig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
...
+	}
+	defer b.bufAllocator.put(strlist)
+	if err := b.args[1].VecEvalString(b.ctx, input, strlist); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(str, strlist)
+	res := result.Int64s()
+	for i := 0; i < n; i++ {
+		strlistI := strlist.GetString(i)
```
Comment (b41sh) [inline expression/builtin_string_vec.go]: check `result.IsNull(i)`

File: expression/builtin_string_vec.go [hunk 4]
```diff
@@ -778,11 +778,43 @@ func (b *builtinLpadSig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
...
+		return err
+	}
+	defer b.bufAllocator.put(strlist)
+	if err := b.args[1].VecEvalString(b.ctx, input, strlist); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(str, strlist)
+	res := result.Int64s()
+	for i := 0; i < n; i++ {
```
Comment (b41sh) [inline expression/builtin_string_vec.go]: check `result.IsNull(i)`

File: expression/builtin_string_vec_test.go [hunk 5]
```diff
@@ -166,7 +166,9 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
 	ast.BitLength: {
 		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETString}},
 	},
-	ast.FindInSet: {},
+	ast.FindInSet: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETString, types.ETString}},
```
Comment (b41sh) [inline expression/builtin_string_vec_test.go]: can we add some testcase to cover strlist is a empty string and str is in the strlist?

Comment (icditwang) [reply expression/builtin_string_vec_test.go]: I added.is it ok?

Comment (b41sh) [reply expression/builtin_string_vec_test.go]: we can use `constStrGener` to generate a given str

Comment (icditwang) [reply expression/builtin_string_vec_test.go]: image
Is it right?

Comment (b41sh) [reply expression/builtin_string_vec_test.go]: right, but a bit too much, I think two is enough,
and we can keep this one
`{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETString, types.ETString}},`

**Comment by b41sh** [comment]
LGTM

**Pull request merged**